tests/core/nogo/custom/foo_func_analyzer.go

```diff
@@ -0,0 +1,37 @@
+// importfmtanalyzer checks for functions named "Foo".
+// It has the same package name as another check to test the checks with
+// the same package name do not conflict.
+package importfmtanalyzer
+
+import (
+	"go/ast"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+const doc = `report calls of functions named "Foo"
+
+The foo_func_analyzer analyzer reports calls to functions that are
+named "Foo".`
+
+var Analyzer = &analysis.Analyzer{
+	Name: "foo_func_analyzer",
+	Run:  run,
+	Doc:  doc,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, f := range pass.Files {
+		ast.Inspect(f, func(n ast.Node) bool {
```

Comment:
In this example, consider using the new "inspect" package rather than ast.Inspect as it is much faster.  It will be checked in to x/tools today and google3 next week.

Reply:
I've filed #1755. Since this is a test though, I'd suggest we proceed with this PR and fix it after we merge the `feature/check` branch into `master` and update `org_golang_x_tools` on master.

Unfortunately, this means we'll need another `go_tool_library` target, but many `Analyzer` implementations will want it.